### PR TITLE
bugfix: using auto adapt batch size with IterBasedRunner

### PR DESCRIPTION
**Pull request by eunwoosh**
### Summary
Fix an issue #2180 
There is no code to handle IterBasedRunner in auto adapt batch size code, so I added it.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ All notable changes to this project will be documented in this file.
 ### Bug fixes
 
 - Fix backward compatibility with OpenVINO SSD-like detection models from OTE 0.5 (<https://github.com/openvinotoolkit/training_extensions/pull/1970>)
+- Fix the bug that auto adapt batch size is unavailable with IterBasedRunner (<https://github.com/openvinotoolkit/training_extensions/pull/2182>)
```
Comment (jaegukhyun) [inline CHANGELOG.md]: Is it right place for changing log?

Comment (eunwoosh) [reply CHANGELOG.md]: Thanks for pointing it! I fixed it.

**Review by jaegukhyun — APPROVED** [review]
LGTM, Thank you for fixing

**Review by sungmanc — APPROVED** [review]
LGTM

**Pull request merged**